circuit-runtime/src/commonMain/kotlin/com/slack/circuit/runtime/Navigator.kt

```diff
@@ -124,12 +125,14 @@ public fun Navigator.popUntil(predicate: (Screen) -> Boolean) {
   while (peek()?.let(predicate) == false) pop() ?: break // Break on root pop
 }
 
-/** Calls [Navigator.pop] until the root screen and passes [result] to the root pop. */
+/** Pop the [Navigator] as if this was the root [Navigator.pop] call. */
 public fun Navigator.popRoot(result: PopResult? = null) {
-  var backStackSize = peekBackStack().size
-  while (backStackSize > 1) {
-    backStackSize--
-    pop()
+  Snapshot.withMutableSnapshot {
+    // Move the top screen to be the root screen for any final animation.
```

Comment:
Can you flesh this out with a little more detail (similar to what you put in the PR description) for posterity?